haystack/document_stores/pinecone.py

```diff
@@ -124,34 +137,32 @@ def __init__(
         self._api_key = api_key
 
         # Formal similarity string
-        if similarity == "cosine":
-            self.metric_type = similarity
-        elif similarity == "dot_product":
-            self.metric_type = "dotproduct"
-        elif similarity in ("l2", "euclidean"):
-            self.metric_type = "euclidean"
-        else:
-            raise ValueError(
-                "The Pinecone document store can currently only support dot_product, cosine and euclidean metrics. "
-                "Please set similarity to one of the above."
-            )
+        self._set_similarity_metric(similarity)
 
         self.similarity = similarity
-        self.index: str = self._index_name(index)
+        self.index: str = self._index(index)
         self.embedding_dim = embedding_dim
         self.return_embedding = return_embedding
         self.embedding_field = embedding_field
         self.progress_bar = progress_bar
         self.duplicate_documents = duplicate_documents
-        self.document_namespace = "no-vectors"
-        self.embedding_namespace = "vectors"
+
+        # Metadata field used for differentiation of:
+        #   - documents with embeddings,
+        #   - documents without embeddings,
+        #   - labels
+        self.type_metadata_field = TYPE_METADATA_FIELD
+        self.document_with_embedding_metadata = DOCUMENT_WITH_EMBEDDING
+        self.document_without_embedding_metadata = DOCUMENT_WITHOUT_EMBEDDING
+        self.label_metadata = LABEL
```

Comment:
Are these meant to be overridable? Cause if they're not it's really not necessary to save them as fields and we can use the constants directly.

Reply:
Basically the same thing you've done for the `OPERATOR` constants.

Reply:
Agree, fixed it now.